projects/igniteui-angular/src/lib/grids/hierarchical-grid/row-island.component.ts

```diff
@@ -90,15 +91,51 @@ export class IgxRowIslandComponent extends IgxHierarchicalGridBaseDirective
     @ContentChildren(IgxColumnComponent, { read: IgxColumnComponent, descendants: false })
     public childColumns = new QueryList<IgxColumnComponent>();
 
+    /**
+     * @hidden
+     */
     @ContentChild(IgxGridToolbarDirective, { read: TemplateRef })
-    public islandToolbarTemplate: TemplateRef<IgxGridToolbarTemplateContext>;
+    protected toolbarDirectiveTemplate: TemplateRef<IgxGridToolbarTemplateContext>;
 
+    /**
+     * @hidden
```

Comment:
Those should really be `/** @hidden @internal **/` though I see there's a pattern in this file already and those are not the only ones

Reply:
Well is there a spec for the @internal use really, because to me it means that it is used only in the component itself.
For typedoc is enough just @hidden to not list it in the API from what I understand.

Reply:
Not really, unfortunately. `@hidden` is to skip something from the API docs and it's strictly for that https://typedoc.org/tags/hidden/, while `@internal` is a TypeScript tag https://typedoc.org/tags/internal/ that can be used to strip those from the type declarations (.d.ts) so those won't be visible in consuming apps intellisense for example. The issue with that is we used on a bunch of cross-component props and or modules for example that we export but don't want in the API and never really turned that option on.

But I digress, this comment was actually irrelevant, since the props are `protected` and need no tags at all lol 😄 You can remove them if you feel like it, since they do nothing anyway

Reply:
It shouldn't affect the cross-component prop use in our source code based on what they say in the TypeScript docs, just the public API in *.d.ts. Anyway that's maybe a topic for another discussion.